dev-docs/source/ISISReflectometryInterface.rst

```diff
@@ -0,0 +1,309 @@
+
+Consider adding new wrapper algorithms if appropriate. :ref:`algm-ReflectometryISISLoadAndProcess` is an algorithm that has been added specifically for this purpose and can usually be extended or modified quite easily because it is designed for use with this GUI. The post-processing algorithm, :ref:`algm-Stitch1DMany`, is more generic so it is likely in future that we would want to add a wrapper for this algorithm rather than changing it directly.
+
+Design pattern examples
+-----------------------
+
+Subscriber pattern
+##################
+
+Let's take the :code:`Event` component as an example.
+
+- The view is constructed first and is passed to the presenter. The presenter then immediately subscribes to the view.
+
+  .. code-block:: c++
+
+    EventPresenter::EventPresenter(IEventView *view)
+        : m_view(view), m_sliceType(SliceType::None) {
+      m_view->subscribe(this);
+    }
+
+- This sets the notifyee in the view, using a subscriber interface.
+
+  .. code-block:: c++
+
+    void QtEventView::subscribe(EventViewSubscriber *notifyee) {
+      m_notifyee = notifyee;
+    }
+
+- The subscriber interface defines the set of notifications that the view needs to send.
+
+  .. code-block:: c++
+
+    class MANTIDQT_ISISREFLECTOMETRY_DLL EventViewSubscriber {
+    public:
+      virtual void notifySliceTypeChanged(SliceType newSliceType) = 0;
+      virtual void notifyUniformSliceCountChanged(int sliceCount) = 0;
+      virtual void notifyUniformSecondsChanged(double sliceLengthInSeconds) = 0;
+      virtual void
+      notifyCustomSliceValuesChanged(std::string pythonListOfSliceTimes) = 0;
+      virtual void
+      notifyLogSliceBreakpointsChanged(std::string logValueBreakpoints) = 0;
+      virtual void notifyLogBlockNameChanged(std::string blockName) = 0;
+    };
+
+- The presenter implements the subscriber interface.
+
+  .. code-block:: c++
+
+    class MANTIDQT_ISISREFLECTOMETRY_DLL EventPresenter
+        : public IEventPresenter,
+          public EventViewSubscriber
+
+- It overrides the notification functions to perform the relevant actions.
+
+  .. code-block:: c++
+
+    void EventPresenter::notifyUniformSliceCountChanged(int) {
+      setUniformSlicingByNumberOfSlicesFromView();
+      m_mainPresenter->notifySettingsChanged();
+    }
+
+- When a user interacts with the view, all the view needs to do is send the appropriate notification. The view does not know anything about the concrete type that it is notifying. This helps to avoid accidentally introducing logic into the view about what should happen on an event and instead just notify that an event happened. It could also be easily extended to support multiple notifyees of different types. 
+
+  .. code-block:: c++
+
+    void QtEventView::onUniformEvenChanged(int numberOfSlices) {
```

Comment:
should `onUniformEvenChanged` be onUniform**Event**Changed?

Reply:
No, it's actually correct - it's not a great name, but `UniformEven` is one of the ways of slicing (by a uniform amount calculated such that all slices are even).